### ISSUE TITLE
ReferenceError: Cannot access 'Object' before initialization

### ISSUE THREAD

**Issue by devuo** [sinclairzx81/typebox#1063]
## Affected version
[0.33.20](https://github.com/sinclairzx81/typebox/releases/tag/0.33.20)

## Description
After updating to typebox version [0.33.20](https://github.com/sinclairzx81/typebox/releases/tag/0.33.20) CommonJS based package failed to start with the following error: `ReferenceError: Cannot access 'Object' before initialization`.

This error does not happen with ESM based packages.

## Steps to reproduce

🔴 CommonJS (Fails):
```
echo "require('@sinclair/typebox')" > reproduce.js
node reproduce.js
```

🟢 ESM (Works):
```
echo "import '@sinclair/typebox'" > reproduce.mjs
node reproduce.mjs
```

## Stack trace

```
~/typebox-test/node_modules/@sinclair/typebox/build/cjs/parse/runtime.js:3
Object.defineProperty(exports, "__esModule", { value: true });
^

ReferenceError: Cannot access 'Object' before initialization
    at Object.<anonymous> (~/typebox-test/node_modules/@sinclair/typebox/build/cjs/parse/runtime.js:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (~/typebox-test/node_modules/@sinclair/typebox/build/cjs/parse/parse.js:7:19)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)

Node.js v20.17.0
```

**Comment by sinclairzx81** [comment]
@devuo Hi, thanks for reporting

There's been a bit of infrastructure added to the project on 0.33.20. Will investigate and report back.

**Comment by sinclairzx81** [comment]
@devuo Hi, thanks again for reporting.

I have published a hotfix on 0.33.21. The issue was related to a variable named `Object` shadowing the `globalThis.Object` (I seem to keep running into this problem unfortunately). The variable has been prefixed with `_Object` inline with code elsewhere.

Apologies for the inconvenience (I really do need better CJS testing)

---

### New Infrastructure

If you're interested, the new infrastructure that's gone in relates to being able to parse TypeScript syntax directly into TB types (instead of using Type.*). You can test with the following.

```typescript
const { Parse } = require('@sinclair/typebox')

const A = Parse(`{ x: number, y: number }`)

console.log(A)
```

The feature is published in an Experimental state currently, but will be expanded upon in 0.34.x. The feature is currently pending documentation, but you can experiment at the link below, I'm currently looking for community feedback on the feature, so open to community insights, bug reports, feature requests to consider for the next main revision.

[TypeScript Link Here](https://www.typescriptlang.org/play/?#code/PTAEFpK6dv4Y+AoEqwBUCeAHApgEID2AHgFygAKAhgE4DOe66ESb7HbyyAlgLY4itAC6gA3lTqMANKADKw6sJ4BjUAF9QAM1pE+oAOQABejwB2KgDbUetYMNx4ARqQPcQrTl+-sWT6qZqDvj0LD7hEbDIKkRm9KIAggD6AAygALySDHgAFAbxtOYA5gYAlNwxcYlJAIwZWYw5AAYFxQDaALpN5dGx8aDJAEz1NNnNAJLCeLRKQgA8ZgCufE7ToAA+oEuWlgB83RV91QDMI1K5TWZ4AO6gOaUZu+LIoKDUACbvFDnUFEsr01kTj+y1WtAe6Se-zB0heoHoi2Bd1+W1BgNASOh0whULRtFhrz4i0s3xRWPxGJBAPBj1R1IJoHePAAbqSqTDKXSwTiuYDkOoDiwAHQi7jBPADVL1BRKVRzcVELSSlK7MWOSV1TIy5QqeWORUa1XIcWS4ZaxQ6vX4A1DI0m5Knc2y3UKpUOo0eSJeiIsRZmHixUDi0Ke71hzi9KqgAhnMYGFxESx4ahmDa82hp1pmEo9E0xp2W13Rj1gcNlrgeSw8KYzSxBxwh0vl5twSP9ADCscaBgARAALPA7Ig9tMAThSE7TWmolkYZTV+FAnYLcqL7ZLnhbW-ALGEixwSfrITC2+3hyjABEuxc2ilZDVZINZMdZAAWWQAVlkADZZAB2Lpc3VK8VxdfUlQvDdTzPDwiCcAArPAVFEYNQAAalAPgiCZLQeGmRtN2gstz36DBr2aMQ4XIdMGSwCgsyKBkAC8KATJMU35QVKn6ORyKaSjXloZN3liSwsDeAB+ejhEKbM4SEj5RPEpEGLhFQpPhGTijhL5NNkopOKAxcyNAq08ANDA7XVXjTKLOQoKI5sWBwOhqD4PAax4Ji8HeI88AIxynLbUQAFEpUyUZGgkMj1FkJoAHk+GrOYMFkAwSAMNMDCwAx9h6bjQtqcjoooSLcgMckyg0OKBNAajUrhOjQAa14WOa-lOkFE0wrSWzwNAHqrMXMLNXkC1V36kaHMC8MWAAazwLADWDE8ZrDYLQAAMWK5rqtAJoFqWpUMC69Vtr660lU26a1q9FhzHePASB8vyAtu70NvGHaYrijA2kOizAIXCUvou8ylXGG73vCFhKiZZRYhnV7VuhmGNoAKW+vamjhMinqmMx3nocQ6vZNZNDhV4JJJoSEUsYQKGy-zMvUSnQAoCRaeJBnDDMIgWeQU7F0xsGDXRqHUa8Fg+GoHB8F81CchiJLs3hIhYnKUNJZ8DaAGksdi-aJDafXzFAAHjo6Cg-t1joNCFiV9dFpVdYl7XfA8P0A1TDDzBrRhkO99DQCKXR920rX3YjArQAAGT4nI4QkajyQ0DYk9AJrU9Zh4ADISba7OHbj6VxrAy646NIA)


Thanks again for reporting the issue
Cheers!
S
